frontend/src/pages/NewRunSwitcher.tsx

```diff
@@ -92,8 +92,8 @@ function NewRunSwitcher(props: PageProps) {
   const pipelineVersionId =
     pipelineVersionIdParam ||
     apiPipeline?.default_version?.id ||
```

Comment:
Would this be changed with the pipeline api integration? I wonder if this change can or should be part of the pipeline api integration change to avoid touching this code twice.

Reply:
If we put this changes in the pipeline API integration, the functionality will break at this moment.

Reply:
Looking back, backend should probably do it in a non-breaking way. Anyway, the fact that this functionality was broken has happened at the time the backend PR merged. Since, apparently we have no test coverage on the feature, which we should improve post-GA, and we are not doing any release until pipeline API integration is done. It's okay to fix it later. 
I don't have a strong opinion on whether to combine or separate the changes, you can make the call.